src/transformers/models/luke/modeling_luke.py

```diff
@@ -0,0 +1,1362 @@
+            details.
+
+        entity_attention_mask (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, entity_length)`, `optional`):
+            Mask to avoid performing attention on padding entity token indices. Mask values selected in ``[0, 1]``:
+
+
+
+            - 1 for entity tokens that are **not masked**,
+            - 0 for entity tokens that are **masked**.
+
+        entity_token_type_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, entity_length)`, `optional`):
+            Segment token indices to indicate first and second portions of the entity token inputs. Indices are
+            selected in ``[0, 1]``:
+
+
+
+            - 0 corresponds to a `portion A` entity token,
+            - 1 corresponds to a `portion B` entity token.
+
+        entity_position_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, entity_length, max_mention_length)`, `optional`):
+            Indices of positions of each input entity in the position embeddings. Selected in the range ``[0,
+            config.max_position_embeddings - 1]``.
+
+        inputs_embeds (:obj:`torch.FloatTensor` of shape :obj:`({0}, hidden_size)`, `optional`):
+            Optionally, instead of passing :obj:`input_ids` you can choose to directly pass an embedded representation.
+            This is useful if you want more control over how to convert :obj:`input_ids` indices into associated
+            vectors than the model's internal embedding lookup matrix.
+
+        head_mask (:obj:`torch.FloatTensor` of shape :obj:`(num_heads,)` or :obj:`(num_layers, num_heads)`, `optional`):
+            Mask to nullify selected heads of the self-attention modules. Mask values selected in ``[0, 1]``:
+
+
+
+            - 1 indicates the head is **not masked**,
+            - 0 indicates the head is **masked**.
+
+        output_attentions (:obj:`bool`, `optional`):
+            Whether or not to return the attentions tensors of all attention layers. See ``attentions`` under returned
+            tensors for more detail.
+        output_hidden_states (:obj:`bool`, `optional`):
+            Whether or not to return the hidden states of all layers. See ``hidden_states`` under returned tensors for
+            more detail.
+        return_dict (:obj:`bool`, `optional`):
+            Whether or not to return a :class:`~transformers.file_utils.ModelOutput` instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The bare LUKE Model transformer outputting raw hidden-states for both word tokens and entities without any specific head on top.",
+    LUKE_START_DOCSTRING,
+)
+class LukeModel(LukePreTrainedModel):
+
+    _keys_to_ignore_on_load_missing = [r"position_ids"]
+
+    def __init__(self, config, add_pooling_layer=True):
+        super().__init__(config)
+        self.config = config
+
+        self.embeddings = LukeEmbeddings(config)
+        self.embeddings.token_type_embeddings.requires_grad = False
```

Comment:
Yes, could you explain? If it's a non-trainable parameter, we should use something different than an `nn.Parameter` for this weight.

Reply:
This can be safely removed. The original purpose of adding this here is to prevent weight updates on this embedding, since the original RoBERTa does not have `token_type_embeddings`.

Reply:
I can remove this line, but LUKE does have token_type_embeddings (for both tokens and entities), right?

Reply:
I am sorry for the delayed reply. LUKE has `token_type_embeddings` both for words and entities. I think removing this line (`self.embeddings.token_type_embeddings.requires_grad = False`) gives nearly no effects to downstream tasks. Also, if I understand correctly, RoBERTa does not have `token_type_embeddings`, but the embedding is created with a zero vector and learned during the fine-tuning task (see [here](https://github.com/huggingface/transformers/blob/master/src/transformers/models/roberta/modeling_roberta.py#L79)).